src/main/sensors/battery.c

```diff
@@ -248,6 +254,112 @@ static void batteryUpdateVoltageState(void)
 
 }
 
+////////
+#ifdef USE_VBAT_ALARM_DELAY
+			//			https ://github.com/betaflight/betaflight/issues/1887
+			//			https ://github.com/betaflight/betaflight/pull/6691
+// uses unfilterd 
+static void batteryUpdateVoltageStateDelay(timeUs_t currentTimeUs)
+{
+	// alerts are currently used by beeper, osd and other subsystems
+	switch (voltageState) {
+	case BATTERY_OK:
+		if (getOptionBatteryVoltage() <= (batteryWarningVoltage - batteryConfig()->vbathysteresis)) {
+
+			if (batteryConfig()->vbat_warn_delay > 0) { // if we have a vbat_warn_delay time 
+
+
+				if (warn_alarm_timeout != 0) {
+					//  check latest battery  for recovery 
+					if ( getOptionBatteryVoltage()   > ( (batteryWarningVoltage - batteryConfig()->vbathysteresis ) + batteryConfig()->vbat_high_hysteresis)) {
```

Comment:
Rules for whitespace are that operators should be surrounded by **one** space, except on the inside of braces.